### PR TITLE
Feature/scaffold

### PR DESCRIPTION
**Pull request by emclaug2**
Changes proposed in this Pull Request:
- This adds the px blue drawer and scaffolds the mono repo using the pxb cli.
- Adds lint (not linted), prettier

This needs a new ng beta package to be published before this can be merged.

### CODE REVIEW

File: src/app/pages/action-list/action-list.component.html [hunk 1]
```diff
@@ -1,29 +1,35 @@
...
-		</p>
-		<button data-cy="action-menu" mat-icon-button #menuTrigger="matMenuTrigger" [matMenuTriggerFor]="menu" (click)="menuTrigger.openMenu()">
-			<mat-icon>more_vert</mat-icon>
-		</button>
-		<mat-menu #menu="matMenu">
-			<button data-cy="remove" mat-menu-item (click)="onRemoveItem(i)">Remove</button>
-			<button data-cy="details" mat-menu-item>View Details</button>
-		</mat-menu>
-	</mat-list-item>
+    <mat-list-item *ngFor="let i of data" [ngClass]="{ selected: isSelected(i) }" (click)="onSelected(i)">
```
Comment (joebochill) [inline src/app/pages/action-list/action-list.component.html:8]: pxb-info-list-item?

File: src/app/pages/action-list/action-list.component.html [hunk 2]
```diff
@@ -1,29 +1,35 @@
...
-	<h1>No Items found </h1>
-	<button data-cy="add-item" mat-raised-button (click)="onAddItem()">
-		Add An Item
-	</button>
-</div>
\ No newline at end of file
+    <h1>No Items found</h1>
+    <button data-cy="add-item" mat-raised-button (click)="onAddItem()">
+        Add An Item
+    </button>
```
Comment (joebochill) [inline src/app/pages/action-list/action-list.component.html]: pxb-empty-state?

File: src/app/pages/app-bar/app-bar.component.html [hunk 3]
```diff
@@ -1,43 +1,59 @@
...
-    <mat-icon mat-list-avatar>person</mat-icon>
-    <div mat-line>{{item.president}}</div>
-    <div mat-line>{{item.party}}</div>
-    <div mat-line>{{item.took_office}}</div>
-    </mat-list-item>
-  </mat-list>
-</div>
\ No newline at end of file
+    <mat-list>
+        <mat-list-item *ngFor="let item of list" style="padding: 16px;">
```
Comment (joebochill) [inline src/app/pages/app-bar/app-bar.component.html:52]: pxb-info-list-item?

File: src/app/pages/app-bar/app-bar.component.ts [hunk 4]
```diff
@@ -4,52 +4,48 @@ import { map } from 'rxjs/operators';
...
-      .pipe(map((event: CdkScrollable) => this.getScrollPosition(event)))
-      .subscribe(scrollTop => this.ngZone.run(() => this.setScroll(scrollTop)));
-  }
-
-  ngAfterViewInit() {
-    if(this.wrapper) {
-      this.width = this.wrapper.nativeElement.clientWidth;
+    @ViewChild('wrapper') wrapper: ElementRef;
+
+    list = presidents;
```
Comment (joebochill) [inline src/app/pages/app-bar/app-bar.component.ts:14]: having the presidents list in a shared folder breaks the modularity of the examples (copy-pastability). I think each example should maintain (a copy of) their own data that they need.

File: src/app/pages/app-bar/app-bar.component.ts [hunk 5]
```diff
@@ -4,52 +4,48 @@ import { map } from 'rxjs/operators';
 import { ScrollDispatcher, CdkScrollable } from '@angular/cdk/scrolling';
 
 @Component({
-  selector: 'app-app-bar',
-  templateUrl: './app-bar.component.html',
-  styleUrls: ['./app-bar.component.scss']
+    selector: 'app-app-bar',
+    templateUrl: './app-bar.component.html',
```
Comment (joebochill) [inline src/app/pages/app-bar/app-bar.component.ts:8]: should this be called 'collapsible-app-bar'?

File: src/app/pages/data-list/data-list.component.html [hunk 6]
```diff
@@ -1,13 +1,13 @@
 <mat-toolbar color="primary">
     <div>Key Value List</div>
-    <div fxFlex style="margin:auto;"></div>
+    <div fxFlex style="margin: auto;"></div>
 </mat-toolbar>
 <mat-list>
-    <mat-list-item *ngFor="let item of list | keyValue;">
-        <div fxLayout="row" style="width:100%">
+    <mat-list-item *ngFor="let item of list | keyValue">
```
Comment (joebochill) [inline src/app/pages/data-list/data-list.component.html:6]: pxb-info-list-item?

**Comment by joebochill** [comment]
Can we also pull in the CircleCI config file from Huayun's PR #3 and close that in favor of this one?

File: src/app/pages/action-list/action-list.component.html (hunk 1 above)
Comment (emclaug2) [reply src/app/pages/action-list/action-list.component.html:8]: Removed the action list.

File: src/app/pages/action-list/action-list.component.html (hunk 2 above)
Comment (emclaug2) [reply src/app/pages/action-list/action-list.component.html]: Removed the action list.

File: src/app/pages/app-bar/app-bar.component.html (hunk 3 above)
Comment (emclaug2) [reply src/app/pages/app-bar/app-bar.component.html:52]: Removed the app bar.

File: src/app/pages/app-bar/app-bar.component.ts (hunk 4 above)
Comment (emclaug2) [reply src/app/pages/app-bar/app-bar.component.ts:14]: Removed the action list.

File: src/app/pages/app-bar/app-bar.component.ts (hunk 5 above)
Comment (emclaug2) [reply src/app/pages/app-bar/app-bar.component.ts:8]: Removed the app bar.

File: src/app/pages/data-list/data-list.component.html (hunk 6 above)
Comment (emclaug2) [reply src/app/pages/data-list/data-list.component.html:6]: I've updated this example to use info list item and found a style bug that only affects pxb-info-list-items used in ng v9.  Made a PR to patch the style.

**Comment by emclaug2** [comment]
> Can we also pull in the CircleCI config file from Huayun's PR #3 and close that in favor of this one?

Yep! I'll add the circleci config into this project.

**Comment by emclaug2** [comment]
> We don't need to include all of these examples in the _scaffold_ story if you don't want to (we have separate stories for each example to bring it over and refactor it). If they're going to be in here though, they should be updated. I would pull them all out except for a simple one (like data list) and leave the refactoring for the other stories or this is going to be a big effort.
> 
> I'm also not so sure about the /data and /shared folders...in a normal application, totally makes sense, but since the intent here is for someone to be able to take an example folder and drop in in their application with minimal/no additional hook-up necessary, we may want to think about refactoring those pieces.

I agree, I just took the existing code that was originally in this repo and just relocated it.  I think the best approach would be to pull and refactor one design pattern at a time.  I've updated the scaffold to only have the data-list design pattern in it.

**Comment by emclaug2** [comment]
This package now includes the circleci config file found in @huayunh 's PR. 
This project is not linted yet because our eslint-config profile needs to be updated to adjust the rules enabled by ^3.0.0 of typescript/eslint-config rules.

**Comment by emclaug2** [comment]
@joebochill  Can you configure circleci to start running for this project?  Only the ci admin can set this up.

**Comment by joebochill** [comment]
@emclaug2 CircleCI is now set up to run checks for this repo

**Pull request merged**